zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -708,7 +708,16 @@ public long getStorageSize() {
         private boolean goToNextLog() throws IOException {
             if (storedFiles.size() > 0) {
                 this.logFile = storedFiles.remove(storedFiles.size() - 1);
-                ia = createInputArchive(this.logFile);
+                try {
+                    ia = createInputArchive(this.logFile);
+                } catch (EOFException e) {
+                    if (storedFiles.isEmpty()) {
+                        LOG.warn("The last log file {} EOF when create input archive, delete it.", logFile.getName(), e);
+                        logFile.deleteOnExit();
```

Comment:
This is somewhat risky. It is not guaranteed that the deletion will ever happen. So it is possible this could introduce partially written file in median of log files.